cpg_pipes/providers/status.py

```diff
@@ -59,3 +63,39 @@ def add_updaters_jobs(
         """
         Add Hail Batch jobs that update the analysis status.
         """
+
+    def slack_env(self, j: Job):
+        """
+        Add environment variables that configure Slack reporter.
+        """
+        if self.slack_channel and self.slack_token:
+            j.env('SLACK_CHANNEL', self.slack_channel)
+            j.env('SLACK_TOKEN', self.slack_token)
+
+    def slack_message_cmd(
+        self,
+        text: str | None = None,
+        data: dict[str, str] | None = None,
+    ) -> str:
+        """
+        Make a bash command that prepares and sends a Slack message.
+        Message can be constructed from a dict `data`, or can be passed directly 
+        as text (`text`). In either case, strings can use Slack `mrkdwn` 
+        for formatting: https://api.slack.com/reference/surfaces/formatting
+        """
+        if not self.slack_channel or not self.slack_token:
```

Comment:
Should this be an assertion / raise an error instead of returning an empty string? (Same below.)